addons/binding/org.openhab.binding.sensebox/src/main/java/org/openhab/binding/sensebox/internal/ExpiringCache.java

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.sensebox.internal;
+
+import java.io.IOException;
+
+/**
+ * This is a simple cache implementation of a single key-value-pair.
+ *
+ * There must be provided an action in order to retrieve/calculate the value for any given key. This action will be
+ * called only if the answer from the last calculation is not valid anymore, i.e. if the key is different or the last
+ * result is expired.
+ *
+ * @author Simon Kaufmann - Initial contribution and API.
+ *
+ * @param
+ *            <Q>the type of the key
+ * @param <V> the type of the value
+ */
+public class ExpiringCache<Q, V> {
```

Comment:
If this is something useful, would it make sense to put it into the ESH core bundles as it is not a all binding related? Where exactly did you copy it from?